domain/secret/state/state.go

```diff
@@ -79,6 +79,11 @@ func (st State) CreateUserSecret(ctx context.Context, version int, uri *coresecr
 		return errors.Trace(err)
 	}
 
+	modelUUID, err := st.GetModelUUID(ctx)
```

Comment:
Can we fetch it in the `grantSecretOwnerManage` in same transaction or even the same query?

Reply:
There's no need to as model uuid will never change and there's trigger on the model table preventing updates. I could add a helper method which takes a txn and call that from here and from GetModelUUID()

Reply:
Added a helper